src/shared/components/palette/organisation/write-organisation-field.component.ts

```diff
@@ -125,12 +125,30 @@ export class WriteOrganisationFieldComponent extends AbstractFieldWriteComponent
     }
   }
 
+  // The way the search works divide into two phases
+  // 1. go through collection of org items one by one by doing the comparsion of search string using includes to all the address fields
+  // 2. split the search string into arrays and apply the each array item into the address fields
+  // 3. both step 1, 2 will go until max count result reaches, and finally combine both result sets into final collection
   public searchOrg(organisations: OrganisationVm[], lowerOrgSearchText: string): SimpleOrganisationModel[] {
-    return organisations.filter(organisation => {
-      return this.searchCriteria(organisation, lowerOrgSearchText) || this.searchWithSpace(organisation, lowerOrgSearchText);
-    })
-      .map(organisation => this.organisationConverter.toSimpleOrganisationModel(organisation))
-      .slice(0, WriteOrganisationFieldComponent.MAX_RESULT_COUNT);
+    let partMatchingResultSet = [], withSpaceMatchingResultSet = [];
+    const MAX_RESULT_COUNT = WriteOrganisationFieldComponent.MAX_RESULT_COUNT
+    organisations.forEach((organisation) => {
+      if ( partMatchingResultSet.length < MAX_RESULT_COUNT && this.searchCriteria(organisation, lowerOrgSearchText)) {
+        partMatchingResultSet.push(organisation);
+      }
+    });
+    organisations.forEach((org) => {
+      const matchingOrg = [...partMatchingResultSet, ...withSpaceMatchingResultSet]
+                          .find(item => item.organisationIdentifier === org.organisationIdentifier)
+      if (!matchingOrg && [...partMatchingResultSet, ...withSpaceMatchingResultSet].length < MAX_RESULT_COUNT
+        && this.searchWithSpace(org, lowerOrgSearchText)
+      ) {
+        withSpaceMatchingResultSet.push(org);
+      }
+    });
+    return [...partMatchingResultSet, ...withSpaceMatchingResultSet].map((organisation) =>
+      this.organisationConverter.toSimpleOrganisationModel(organisation)
+    );
```

Comment:
This seems like only for exactly search for the first time, if nothing is found then do the fuzzy search with space, if found something don't do the fuzzy search with space.
And two forEach here seems not necessary, probably there is potentially some performance issue occur.
As chat yesterday I think the easier way is first verify whether the search string is a post code pattern using Regex, if yes remove the space in the search string and just apply the exactly search here then it will be easier. The original searchWithSpace purpost is actually for searching the postcode. 
It will be good to retest this ticket: https://tools.hmcts.net/jira/browse/EUI-2087 for this change.
But search is a big topic, it depends whether user wants this searchWithSpace, if they don't like it we can provide search with exact word, that will get everything to be easier.

Reply:
@rayliangatos discussed a few options with Andy on the last design drop session and we have briefly agreed of going this way in a simpler way instead of redoing the whole stuff.